src/main/java/co/rsk/federate/btcreleaseclient/cache/PegoutSignedCacheFactory.java

```diff
@@ -0,0 +1,48 @@
+package co.rsk.federate.btcreleaseclient.cache;
+
+import co.rsk.panic.PanicProcessor;
+import java.time.Duration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PegoutSignedCacheFactory {
+
+  private static final Logger logger = LoggerFactory.getLogger(PegoutSignedCacheFactory.class);
+  private static final PanicProcessor panicProcessor = new PanicProcessor();
+
+  private static PegoutSignedCache instance;
+
+  private PegoutSignedCacheFactory() {
+  }
+
+  /**
+   * Retrieves an instance of PegoutSignedCache with the specified Time To Live
+   * (TTL). If an instance does not already exist, it is lazily initialized.
+   * 
+   * @param ttl The duration for which the cache entries are considered valid.
+   * @return An instance of PegoutSignedCache.
+   * @throws IllegalArgumentException If the supplied TTL value is invalid.
+   */
+  public static PegoutSignedCache getInstance(final Duration ttl) throws IllegalArgumentException {
```

Comment:
maybe directly call the variable `timeToLive` ?
So we don't have to check the documentation to understand the name

Reply:
`ttl` is a common way to identify "time to live" especially in a caching context, lets keep it as is.

Reply:
didnt know, ok!